src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -56,8 +56,7 @@ export const _UserAvatar: React.FC<UserAvatarProps> = ({
 
   const containerClasses = classNames("UserAvatar", containerClassName, {
     "UserAvatar--clickable": onClick !== undefined,
-    "UserAvatar--large": large,
-    "UserAvatar--medium": medium,
+    [`UserAvatar--${size}`]: size,
```

Comment:
Size is limited to a handful values, it shouldn't be a problem either being undefined or `"legacy"`, `"small"` etc.